django_ipam/base/models.py

```diff
@@ -1,28 +1,68 @@
+from ipaddress import ip_address, ip_network
+
 import swapper
+from django.conf import settings
+from django.core.exceptions import ValidationError
 from django.db import models
+from django.utils.translation import gettext_lazy as _
 from openwisp_utils.base import TimeStampedEditableModel
 
+from .fields import NetworkField
+
 
 class AbstractSubnet(TimeStampedEditableModel):
-    name = models.CharField(max_length=100, blank=True)  # Add a default value
+    subnet = NetworkField(default="10.0.0.0/24", db_index=True)
     description = models.CharField(max_length=100, blank=True)
+    master_subnet = models.ForeignKey('self', on_delete=models.CASCADE,
+                                      blank=True, null=True,
+                                      related_name="child_subnets")
 
     class Meta:
         abstract = True
+        indexes = [
+            models.Index(fields=['subnet'], name='subnet_idx')
+        ]
 
     def __str__(self):
-        return self.name
+        return str(self.subnet)
+
+    def clean_fields(self, exclude=None):
+        super().clean_fields(exclude=exclude)
+        for subnet in swapper.load_model("django_ipam", "Subnet").objects.filter().values():
+            if(ip_network(self.subnet).overlaps(subnet["subnet"])):
+                raise ValidationError({'subnet': _('Subnet overlaps with %s') % (subnet["subnet"])})
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
```

Comment:
Remove this, never ever do this. In django, model validation is performed explicitly. If you add this here, in many other parts django will execute the validation twice